python/src/trezorlib/device.py

```diff
@@ -220,6 +220,20 @@ def cancel_authorization(client: "TrezorClient") -> "MessageType":
     return client.call(messages.CancelAuthorization())
 
 
+@expect(messages.UnlockedPathRequest, field="mac", ret_type=bytes)
+def unlock_path(client: "TrezorClient", n: "Address") -> "MessageType":
+    resp = client.call(messages.UnlockPath(address_n=n))
+
+    # Cancel the UnlockPath workflow now that we have the authentication code.
+    cancel_resp = client.call_raw(messages.Cancel())
+    if not isinstance(cancel_resp, messages.Failure):
+        raise RuntimeError("Invalid response")
+    if cancel_resp.code != messages.FailureType.ActionCancelled:
+        raise TrezorFailure(cancel_resp)
```

Comment:
```suggestion
    try:
        client.cancel()
    except exceptions.Cancelled:
        return resp
    else:
        raise TrezorException("Unexpected response in UnlockPath flow")
```

Reply:
Fixup needed https://github.com/trezor/trezor-firmware/pull/2289/commits/c421a7fe61c7124419ca79a3727bdc40f63af212, since `client.cancel()` does not wait for the response.